### PR TITLE
docs: document `PIXEL.Avatar`

### PR DESCRIPTION
**Pull request by Gravy59**
References `PIXEL.Avatar` task in #6.
Please squash these commits!

### CODE REVIEW

File: components/realm/index.jsx [hunk 1]
```diff
@@ -20,6 +20,10 @@ export default function Realm({ type, size = 4 }) {
 			background = "linear-gradient(45deg, #dea909 53%, #03a9f4 53.001%)";
 			name = "Client & Server";
 			break;
+		case "shared-menu":
+			background = "linear-gradient(45deg, #dea909 53%, #4caf50 53.001%)";
+			name = "Client & Menu";
+			break;
```
Comment (TomDotBat) [inline components/realm/index.jsx]: Not necessary, PIXEL UI does not officially support the menu state.

File: pages/pixel-ui/_meta.json [hunk 2]
```diff
@@ -11,6 +11,11 @@
 	"formatting": "Formatting",
 	"loaders": "Loaders",
 	"scaling": "Scaling",
+	"--vgui": {
+		"title": "VGUI Reference",
+		"type": "separator"
+	},
```
Comment (TomDotBat) [inline pages/pixel-ui/_meta.json]: I wouldn't put this under its own separator. Instead, create a category page under the API Reference section with the name "Elements". This page should mention that elements can be created with `vgui.Create`  and include the full list of documented elements below.

File: pages/pixel-ui/avatar.mdx [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+		href: "avatar/set-steamid"
+	},
+	{
+		name: "SetPlayer",
+		parent: "Avatar:",
+		description: "Loads avatar for a given player.",
+		args: ["Player", "number"],
+		href: "avatar/set-player"
+	},
+
```
Comment (TomDotBat) [inline pages/pixel-ui/avatar.mdx]: Remove the trailing comma and whitespace.

File: pages/pixel-ui/avatar.mdx [hunk 4]
```diff
@@ -0,0 +1,22 @@
+import MethodList from "/components/method_list";
+
+# Avatar
+## Methods
+The following vgui methods are accessible via the `PANEL` table:
```
Comment (TomDotBat) [inline pages/pixel-ui/avatar.mdx]: This should say the following:
The following methods are accessible through the `Avatar` element table:

File: pages/pixel-ui/avatar.mdx [hunk 5]
```diff
@@ -0,0 +1,22 @@
+import MethodList from "/components/method_list";
+
+# Avatar
+## Methods
```
Comment (TomDotBat) [inline pages/pixel-ui/avatar.mdx]: A new section should be added above the methods which details the Panel which this element derives from. See the following page under the "Parent" section as an example: https://wiki.facepunch.com/gmod/DLabel#Parent

File: pages/pixel-ui/avatar.mdx (hunk 3 above)
Comment (TomDotBat) [inline pages/pixel-ui/avatar.mdx]: Get/SetMaskSize should also be documented, see the following source:
https://github.com/TomDotBat/pixel-ui/blob/master/lua/pixelui/elements/cl_avatar.lua#L20

File: pages/pixel-ui/avatar/set-player.mdx [hunk 6]
```diff
@@ -0,0 +1,17 @@
+import Method from "/components/method";
+import Argument from "/components/method/argument";
+import ReturnValue from "/components/method/return_value";
+
+# SetPlayer
+<Method
+	name="Avatar:SetPlayer"
+	realm="shared-menu"
```
Comment (TomDotBat) [inline pages/pixel-ui/avatar/set-player.mdx]: Should be the client realm.

File: pages/pixel-ui/avatar/set-steamid.mdx [hunk 7]
```diff
@@ -0,0 +1,17 @@
+import Method from "/components/method";
+import Argument from "/components/method/argument";
+import ReturnValue from "/components/method/return_value";
+
+# SetSteamID
+<Method
+	name="Avatar:SetSteamID"
+	realm="shared-menu"
```
Comment (TomDotBat) [inline pages/pixel-ui/avatar/set-steamid.mdx]: Should be the client realm.

File: pages/pixel-ui/avatar/set-steamid.mdx [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+	name="Avatar:SetSteamID"
+	realm="shared-menu"
+	args={[
+		<Argument type="string" name="id">The 64bit SteamID of the player to load avatar of</Argument>,
+		<Argument type="number" name="size">The size of the avatar to use. Acceptable sizes are 32, 64, 184.</Argument>
+	]}
+	repositoryUrl="https://github.com/TomDotBat/pixel-ui"
+	declaration="lua/pixelui/elements/cl_avatar.lua#L47"
+>
+    Loads avatar by its 64-bit Steam ID (community ID).
```
Comment (TomDotBat) [inline pages/pixel-ui/avatar/set-steamid.mdx]: Add this link to "Steam ID": https://developer.valvesoftware.com/wiki/SteamID

Comment (TomDotBat) [inline pages/pixel-ui/avatar/set-steamid.mdx]: Should be:
Loads **an** avatar by a 64-bit SteamID (community ID).

File: pages/pixel-ui/avatar/set-player.mdx [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+	name="Avatar:SetPlayer"
+	realm="shared-menu"
+	args={[
+		<Argument type="Player" name="ply">The player to use avatar of.</Argument>,
+		<Argument type="number" name="size">The size of the avatar to use. Acceptable sizes are 32, 64, 184.</Argument>
+	]}
+	repositoryUrl="https://github.com/TomDotBat/pixel-ui"
+	declaration="lua/pixelui/elements/cl_avatar.lua#L43"
+>
+	Loads avatar for a given player.
```
Comment (TomDotBat) [inline pages/pixel-ui/avatar/set-player.mdx]: Should be:
Loads the avatar of a given player.

File: pages/pixel-ui/avatar/set-player.mdx [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+# SetPlayer
+<Method
+	name="Avatar:SetPlayer"
+	realm="shared-menu"
+	args={[
+		<Argument type="Player" name="ply">The player to use avatar of.</Argument>,
+		<Argument type="number" name="size">The size of the avatar to use. Acceptable sizes are 32, 64, 184.</Argument>
+	]}
+	repositoryUrl="https://github.com/TomDotBat/pixel-ui"
+	declaration="lua/pixelui/elements/cl_avatar.lua#L43"
```
Comment (TomDotBat) [inline pages/pixel-ui/avatar/set-player.mdx]: Usage of multi-line selections is preferred for adding method declaration links, this one should be `#L43-L45`.

File: pages/pixel-ui/avatar/set-player.mdx [hunk 11]
```diff
@@ -0,0 +1,17 @@
+import Method from "/components/method";
+import Argument from "/components/method/argument";
+import ReturnValue from "/components/method/return_value";
+
+# SetPlayer
+<Method
+	name="Avatar:SetPlayer"
+	realm="shared-menu"
+	args={[
+		<Argument type="Player" name="ply">The player to use avatar of.</Argument>,
```
Comment (TomDotBat) [inline pages/pixel-ui/avatar/set-player.mdx]: The player to **the** use avatar of.

File: pages/pixel-ui/avatar/set-player.mdx [hunk 12]
```diff
@@ -0,0 +1,17 @@
...
+import Argument from "/components/method/argument";
+import ReturnValue from "/components/method/return_value";
+
+# SetPlayer
+<Method
+	name="Avatar:SetPlayer"
+	realm="shared-menu"
+	args={[
+		<Argument type="Player" name="ply">The player to use avatar of.</Argument>,
+		<Argument type="number" name="size">The size of the avatar to use. Acceptable sizes are 32, 64, 184.</Argument>
```
Comment (TomDotBat) [inline pages/pixel-ui/avatar/set-player.mdx]: Should be:
The size of the avatar, must be 32, 64 or 184.

File: pages/pixel-ui/avatar/set-steamid.mdx [hunk 13]
```diff
@@ -0,0 +1,17 @@
...
+import Argument from "/components/method/argument";
+import ReturnValue from "/components/method/return_value";
+
+# SetSteamID
+<Method
+	name="Avatar:SetSteamID"
+	realm="shared-menu"
+	args={[
+		<Argument type="string" name="id">The 64bit SteamID of the player to load avatar of</Argument>,
+		<Argument type="number" name="size">The size of the avatar to use. Acceptable sizes are 32, 64, 184.</Argument>
```
Comment (TomDotBat) [inline pages/pixel-ui/avatar/set-steamid.mdx]: Should be:
The size of the avatar, must be 32, 64 or 184.

File: pages/pixel-ui/avatar/set-steamid.mdx [hunk 14]
```diff
@@ -0,0 +1,17 @@
+import Method from "/components/method";
+import Argument from "/components/method/argument";
+import ReturnValue from "/components/method/return_value";
+
+# SetSteamID
+<Method
+	name="Avatar:SetSteamID"
+	realm="shared-menu"
+	args={[
+		<Argument type="string" name="id">The 64bit SteamID of the player to load avatar of</Argument>,
```
Comment (TomDotBat) [inline pages/pixel-ui/avatar/set-steamid.mdx]: Keep consistency:
64bit -> 64-bit

File: pages/pixel-ui/avatar/set-steamid.mdx [hunk 15]
```diff
@@ -0,0 +1,17 @@
...
+# SetSteamID
+<Method
+	name="Avatar:SetSteamID"
+	realm="shared-menu"
+	args={[
+		<Argument type="string" name="id">The 64bit SteamID of the player to load avatar of</Argument>,
+		<Argument type="number" name="size">The size of the avatar to use. Acceptable sizes are 32, 64, 184.</Argument>
+	]}
+	repositoryUrl="https://github.com/TomDotBat/pixel-ui"
+	declaration="lua/pixelui/elements/cl_avatar.lua#L47"
```
Comment (TomDotBat) [inline pages/pixel-ui/avatar/set-steamid.mdx]: Usage of multi-line selections is preferred for adding method declaration links, this one should be `#L47-L49`.

File: pages/pixel-ui/avatar/set-steamid.mdx [hunk 16]
```diff
@@ -0,0 +1,17 @@
+import Method from "/components/method";
+import Argument from "/components/method/argument";
+import ReturnValue from "/components/method/return_value";
```
Comment (TomDotBat) [inline pages/pixel-ui/avatar/set-steamid.mdx]: Remove this unnecessary import.

File: pages/pixel-ui/avatar/set-player.mdx [hunk 17]
```diff
@@ -0,0 +1,17 @@
+import Method from "/components/method";
+import Argument from "/components/method/argument";
+import ReturnValue from "/components/method/return_value";
```
Comment (TomDotBat) [inline pages/pixel-ui/avatar/set-player.mdx]: Remove this unnecessary import.

**Review by TomDotBat — CHANGES_REQUESTED** [review]

**Comment by Gravy59** [comment]
Thanks, I'll use these as the standard for VGUI elements from now on

**Pull request closed without merging**